proxyd/consensus_poller.go

```diff
@@ -306,30 +323,56 @@ func (cp *ConsensusPoller) UpdateBackend(ctx context.Context, be *Backend) {
 	RecordConsensusBackendInSync(be, err == nil && inSync)
 	if err != nil {
 		log.Warn("error updating backend sync state", "name", be.Name, "err", err)
+		return
 	}
 
 	var peerCount uint64
 	if !be.skipPeerCountCheck {
 		peerCount, err = cp.getPeerCount(ctx, be)
 		if err != nil {
 			log.Warn("error updating backend peer count", "name", be.Name, "err", err)
+			return
+		}
+		if peerCount == 0 {
+			log.Warn("peer count responded with 200 and 0 peers", "name", be.Name)
+			return
 		}
```

Comment:
Just to confirm this is the expected behavior for peer count zero?

Reply:
we can remove this check

```
		if peerCount == 0 {
			log.Warn("peer count responded with 200 and 0 peers", "name", be.Name)
			return
		}
```